<a href="https://colab.research.google.com/github/tastiz/hello-world/blob/master/ksprojects.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [23]:
#Importing the ksprojects table from kickstarterprojects
#using pandas to read sql and use table created

from sqlalchemy import create_engine

postgres_user = 'dabc_student'
postgres_pw = '7*.8G9QH21'
postgres_host = '142.93.121.174'
postgres_port = '5432'
postgres_db = 'kickstarterprojects'

engine = create_engine('postgresql://{}:{}@{}:{}/{}'.format(
    postgres_user, postgres_pw, postgres_host, postgres_port, postgres_db))

import pandas as pd

sql = 'SELECT * FROM ksprojects '

kickstarter_df = pd.read_sql_query(sql, con=engine)

#Returns the dimensions of rows and columns in the imported table
kickstarter_df.shape

#Returns column names and type of data stored within
kickstarter_df.info()

#Finds any null values found in the table
def find_null(kickstarter_df):
  null_values = kickstarter_df.columns[kickstarter_df.isnull().any()]
  kickstarter_df[null_values].isnull().sum()
  
print(find_null(kickstarter_df))

print(kickstarter_df[kickstarter_df.isnull().any(axis=1)][null_columns].head())

#The Average pledged amount was 7900.24 usd
sql3 = 'SELECT AVG(usd_pledged) FROM ksprojects WHERE usd_pledged IS NOT NULL'

average_pledged = pd.read_sql_query(sql3, con=engine)
print(average_pledged)






<class 'pandas.core.frame.DataFrame'>
RangeIndex: 65418 entries, 0 to 65417
Data columns (total 13 columns):
id               65418 non-null int64
name             65418 non-null object
category         65418 non-null object
main_category    65418 non-null object
deadline         65418 non-null datetime64[ns]
goal             65418 non-null float64
launched         65418 non-null datetime64[ns]
state            65418 non-null object
backers          65418 non-null int64
country          65418 non-null object
usd_pledged      64673 non-null float64
currency         65418 non-null object
pledged          65418 non-null float64
dtypes: datetime64[ns](2), float64(3), int64(2), object(6)
memory usage: 6.5+ MB
None
       usd_pledged
9490           NaN
15111          NaN
17484          NaN
17532          NaN
27334          NaN
           avg
0  7900.242005


There are 65418 rows and 13 columns in the ksprojects table



id        :         int64
name       :        object
category     :      object
main_category :     object
deadline       :    datetime64
goal            :   float64
launched        :   datetime64
state           :   object
backers         :   int64
country         :   object
usd_pledged     :  float64
currency           object
pledged        :    float64

There are  null values in the usd_pledged column in ksprojects table
